### PR TITLE
[REL-1405] Release notes for v23.2.13

### PR DESCRIPTION
**Pull request by mdlinville**
[REL-1405] Release notes for v23.2.13

### CODE REVIEW

File: src/current/_includes/releases/v23.2/v23.2.13.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
+## v23.2.13
+
+Release Date: October 17, 2024
+
+{% include releases/new-release-downloads-docker-image.md release=include.release %}
+
+<h3 id="v23-2-13-{{-site.data.products.enterprise-}}-edition-changes">{{ site.data.products.enterprise }} edition changes</h3>
+
+- The description for the [cluster setting]({% link v23.2/cluster-settings.md %}) `changefeed.sink_io_workers` now lists all sinks that support the setting. [#130372][#130372]
```
Comment (mikeCRL) [inline src/current/_includes/releases/v23.2/v23.2.13.md:9]:
```suggestion
- The description for the [cluster setting]({% link v23.2/cluster-settings.md %}) `changefeed.sink_io_workers` now lists all [changefeed sinks]({% link v23.2/changefeed-sinks.md %}) that support the setting. [#130372][#130372]
```

File: src/current/_includes/releases/v23.2/v23.2.13.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+- Network metrics have been added for the following [changefeed sinks]({% link v23.2/changefeed-sinks.md %}):
+      - Kafka v1
+      - Webhook
+      - PubSub
+      - SQL
+
+      [#130664][#130664]
```
Comment (mikeCRL) [inline src/current/_includes/releases/v23.2/v23.2.13.md:16]: Consider the detail I took from Ryan's edit of the same backport https://github.com/cockroachdb/docs/blob/958ad9a2b9ed8f9d26df8c6ce13e53955f3f4eaf/src/current/_includes/releases/v24.2/v24.2.4.md?plain=1#L10-L16

File: src/current/_includes/releases/v23.2/v23.2.13.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+- The new [metric]({% link v23.2/metrics.md %}) `changefeed.total_ranges` allows observation of the number of changes that are watched by a changefeed aggregator. It uses the same polling interval as `changefeed.lagging_ranges`, which is controlled by the changefeed option `lagging_ranges_polling_interval`. [#130984][#130984]
+- The following groups of [metrics]({% link v23.2/metrics.md %}) and [logs]({% link v23.2/logging.md %}) have been renamed to include the buffer they are associated with. The previous metrics are still maintained for backward compatibility.
+      - `changefeed.buffer_entries_mem.*`
+      - `changefeed.buffer_pushback_nanos.*`
```
Comment (mikeCRL) [inline src/current/_includes/releases/v23.2/v23.2.13.md:21]: I think you inadvertently dropped `changefeed.buffer_entries.*` from the original RN

File: src/current/_includes/releases/v23.2/v23.2.13.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+      [#131417][#131417]
+- Timers have been added to key parts of the [changefeed]({% link v23.2/change-data-capture-overview.md %}) pipeline to help debug problems with feeds. [Metrics]({% link v23.2/metrics.md %}) `changefeed.stage.{stage}.latency` now emit latency histograms for each stage. The changefeed `scope` label allows debugging of a specific changefeed. [#131428][#131428]
```
Comment (mikeCRL) [inline src/current/_includes/releases/v23.2/v23.2.13.md:24]: Grammar issue in second sentence. I like the version you helped edit here

https://github.com/cockroachdb/docs/blob/d9c042a7c68b28dfc2736655954936c7ca65b1c2/src/current/_includes/releases/v24.2/v24.2.4.md?plain=1#L19

**Review by mikeCRL — COMMENTED** [review]
Partial review. Continuing.

File: src/current/_includes/releases/v23.2/v23.2.13.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+- The following new [metrics]({% link v23.2/metrics.md %}) track the number of bytes sent by an individual [changefeed]({% link v23.2/change-data-capture-overview.md %}) to each sink:
+    - `changefeed.network.bytes_in`
+    - `changefeed.network.bytes_out`
+
+    [#130664][#130664]
+- You can now set the log format for the `STDERR` changefeed sink using the `format` field in the `stderr` sink section of the [logging]({% link v23.2/logging.md %}) configuration. [#131533][#131533]
+
+<h3 id="v23-2-13-db-console-changes">DB Console changes</h3>
+
+- The [DB Console]({% link v23.2/ui-overview.md %}) now shows an alert message if the cluster's [enterprise license]({% link v23.2/licensing-faqs.md %}) is expired or will expire in fewer than 15 days, or if the cluster has no enterprise license set. [#130425][#130425]
```
Comment (mikeCRL) [inline src/current/_includes/releases/v23.2/v23.2.13.md:46]: Our edit from my PR for consideration https://github.com/cockroachdb/docs/blob/1e52565e2fb84766836bc1824435d7351d4ad6e5/src/current/_includes/releases/v24.2/v24.2.4.md?plain=1#L33

File: src/current/_includes/releases/v23.2/v23.2.13.md [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+{% include releases/new-release-downloads-docker-image.md release=include.release %}
+
+<h3 id="v23-2-13-{{-site.data.products.enterprise-}}-edition-changes">{{ site.data.products.enterprise }} edition changes</h3>
+
+- The description for the [cluster setting]({% link v23.2/cluster-settings.md %}) `changefeed.sink_io_workers` now lists all sinks that support the setting. [#130372][#130372]
+- Network metrics have been added for the following [changefeed sinks]({% link v23.2/changefeed-sinks.md %}):
+      - Kafka v1
+      - Webhook
+      - PubSub
+      - SQL
```
Comment (mdlinville) [inline src/current/_includes/releases/v23.2/v23.2.13.md:14]:
```suggestion
Added two network metrics, `changefeed.network.bytes_in` and `changefeed.network.bytes_out`. [#130578][#130578] 
	  
	   These metrics track the number of bytes sent by individual [changefeeds]({% link v24.2/change-data-capture-overview.md %}) to the following sinks:  
	 	- [Kafka sinks]({% link v24.2/changefeed-sinks.md %}#kafka). If [child metrics are enabled]({% link v24.2/cluster-settings.md %}#setting-server-child-metrics-enabled), the metric will have a `kafka` label.  
	 	- [Webhook sinks]({% link v24.2/changefeed-sinks.md %}#webhook-sink). If child metrics are enabled, the metric will have a `webhook` label.  
	 	- [Pub/Sub sinks]({% link v24.2/changefeed-sinks.md %}#google-cloud-pub-sub). If child metrics are enabled, the metric will have a `pubsub` label.  
	 	- [SQL sink]({% link v24.2/changefeed-for.md %}). If child metrics are enabled, the metric will have a `sql` label. 
```

File: src/current/_includes/releases/v23.2/v23.2.13.md [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+
+<h3 id="v23-2-13-db-console-changes">DB Console changes</h3>
+
+- The [DB Console]({% link v23.2/ui-overview.md %}) now shows an alert message if the cluster's [enterprise license]({% link v23.2/licensing-faqs.md %}) is expired or will expire in fewer than 15 days, or if the cluster has no enterprise license set. [#130425][#130425]
+
+<h3 id="v23-2-13-bug-fixes">Bug fixes</h3>
+
+- Fixed a bug where the command `SHOW CLUSTER SETTING FOR VIRTUAL CLUSTER` would erroneously return `NULL` for some settings. [#128782][#128782]
+- Fixed a bug where a node could fail to start with the error `could not insert session ...: unexpected value` if an ambiguous result error occurred while inserting data into the `sqlliveness` table. [#130343][#130343]
+- Fixed a bug that could prevent [upgrade finalization]({% link v23.2/upgrade-cockroach-version.md %}) due to the upgrade pre-condition for repairing descriptor corruption. [#130519][#130519]
```
Comment (mikeCRL) [inline src/current/_includes/releases/v23.2/v23.2.13.md:52]: Not going to get picky about aligning all of these with my PR, unless it's a very helpful change. And soon enough, we won't need to worry about such divergence across backports.

File: src/current/_includes/releases/v23.2/v23.2.13.md [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+- Fixed a bug that could result in the erroneous log message `expiration of liveness record ... is not greater than expiration of the previous lease ... after liveness heartbeat`. [#130790][#130790]
+- Fixed a bug where queries that are not initiated within a SQL session could fail to respect a statement timeout, including [background jobs]({% link v23.2/jobs.md %}), queries issued by the [DB Console]({% link v23.2/ui-overview.md %}) that perform introspection, and the [CockroachDB {{ site.data.products.cloud }} SQL Shell]({% link cockroachcloud/sql-shell.md %}). [#130790][#130790]
+- Fixed a bug where a connection could be incorrectly dropped if the client was attempting to change a schema at the same time that the same schema's objects were being dropped. [#130964][#130964]
+- Fixed a bug that could cause the following error to be logged when executing a query under [READ COMMITTED]({% link v23.2/read-committed.md %}) isolation if it involved a table with `NOT NULL` [virtual columns]({% link v23.2/virtual.md %}). : `internal error: Non-nullable column ...`. [#131065][#131065]
```
Comment (mikeCRL) [inline src/current/_includes/releases/v23.2/v23.2.13.md:57]:
```suggestion
- Fixed a bug that could cause the following error to be logged when executing a query under [READ COMMITTED]({% link v23.2/read-committed.md %}) isolation if it involved a table with `NOT NULL` [virtual columns]({% link v23.2/virtual.md %}): `internal error: Non-nullable column ...`. [#131065][#131065]
```

File: src/current/_includes/releases/v23.2/v23.2.13.md [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+- Fixed a potential memory leak in [changefeeds]({% link v23.2/change-data-capture-overview.md %}) that use a cloud storage sink. The memory leak could occur if both of the [cluster settings]({% link v23.2/cluster-settings.md %}) `changefeed.fast_gzip.enabled` and `changefeed.cloudstorage.async_flush.enabled` are `true` **and** if the changefeed received an error while attempting to write to the sink. [#130614][#130614]
```
Comment (mikeCRL) [inline src/current/_includes/releases/v23.2/v23.2.13.md:58]:
```suggestion
- Fixed a potential memory leak in [changefeeds]({% link v23.2/change-data-capture-overview.md %}) that use a cloud storage sink. The memory leak could occur if both of the [cluster settings]({% link v23.2/cluster-settings.md %}) `changefeed.fast_gzip.enabled` and `changefeed.cloudstorage.async_flush.enabled` were `true` **and** if the changefeed received an error while attempting to write to the sink. [#130614][#130614]
```

**Review by mikeCRL — APPROVED** [review]
Approving with some additional suggestions/comments.

File: src/current/_includes/releases/v23.2/v23.2.13.md [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+      - Kafka v1
+      - Webhook
+      - PubSub
+      - SQL
+
+      [#130664][#130664]
+
+- The new [metric]({% link v23.2/metrics.md %}) `changefeed.total_ranges` allows observation of the number of changes that are watched by a changefeed aggregator. It uses the same polling interval as `changefeed.lagging_ranges`, which is controlled by the changefeed option `lagging_ranges_polling_interval`. [#130984][#130984]
+- The following groups of [metrics]({% link v23.2/metrics.md %}) and [logs]({% link v23.2/logging.md %}) have been renamed to include the buffer they are associated with. The previous metrics are still maintained for backward compatibility.
+      - `changefeed.buffer_entries_mem.*`
```
Comment (mdlinville) [inline src/current/_includes/releases/v23.2/v23.2.13.md:20]:
```suggestion
      - `changefeed.buffer_entries.*`
      - `changefeed.buffer_entries_mem.*`
```

File: src/current/_includes/releases/v23.2/v23.2.13.md (hunk 4 above)
Comment (mdlinville) [inline src/current/_includes/releases/v23.2/v23.2.13.md:24]:
```suggestion
- Added timers and corresponding [metrics]({% link v24.2/metrics.md %} for key parts of the [changefeed]({% link v24.2/change-data-capture-overview.md %}) pipeline to help debug issues with feeds. The `changefeed.stage.{stage}.latency` metrics now emit latency histograms for each stage. The metrics respect the changefeed `scope` label to debug a specific feed. [#131428][#131428]
```

File: src/current/_includes/releases/v23.2/v23.2.13.md [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+- The following new [metrics]({% link v23.2/metrics.md %}) track the number of bytes sent by an individual [changefeed]({% link v23.2/change-data-capture-overview.md %}) to each sink:
+    - `changefeed.network.bytes_in`
+    - `changefeed.network.bytes_out`
+
+    [#130664][#130664]
+- You can now set the log format for the `STDERR` changefeed sink using the `format` field in the `stderr` sink section of the [logging]({% link v23.2/logging.md %}) configuration. [#131533][#131533]
+
+<h3 id="v23-2-13-db-console-changes">DB Console changes</h3>
+
+- The [DB Console]({% link v23.2/ui-overview.md %}) now shows an alert message if the cluster's [enterprise license]({% link v23.2/licensing-faqs.md %}) is expired or will expire in fewer than 15 days, or if the cluster has no enterprise license set. [#130425][#130425]
```
Comment (mdlinville) [inline src/current/_includes/releases/v23.2/v23.2.13.md:48]:
```suggestion
- The [DB Console]({% link v24.2/ui-overview.md %}) now shows a notification if the cluster has no [Enterprise license]({% link v24.2/enterprise-licensing.md %}) set. Refer to [upcoming license changes](https://www.cockroachlabs.com/enterprise-license-update/) for more information. [#130425][#130425]
```

**Pull request merged**